src/ngx_http_lua_shdict.c

```diff
@@ -56,6 +56,31 @@ static ngx_inline ngx_shm_zone_t *ngx_http_lua_shdict_get_zone(lua_State *L,
 #define NGX_HTTP_LUA_SHDICT_RIGHT       0x0002
 
 
+#ifdef NGX_HTTP_LUA_SHDICT_HAVE_RWLOCK
+#define NGX_HTTP_LUA_SHDICT_RLOCK 0
+#define NGX_HTTP_LUA_SHDICT_WLOCK 1
+
+#define ngx_http_lua_shdict_lock(ctx, t)        \
+    if (t == NGX_HTTP_LUA_SHDICT_RLOCK) {       \
```

Comment:
Please make sure all macro parameters always get enclosed by parentheses to avoid potential operator precedence issues in macro expansion.

Also, I wonder can we make this a static C inline function instead?

Reply:
Yeah, refactored to make a `static ngx_inline void`.